pkgs/applications/office/paperwork/default.nix

```
@@ -46,9 +54,27 @@ python3Packages.buildPythonApplication rec {
     paths = lib.collect lib.isDerivation aspellDicts;
   }}/lib/aspell";
 
+  postInstall = ''
+    # paperwork-shell needs to be re-wrapped with access to paperwork
+    cp ${python3Packages.paperwork-backend}/bin/.paperwork-shell-wrapped $out/bin/paperwork-shell
+    # install desktop files and icons
+    # paperwork-shell install copies everything to $HOME/.local/share instead of $out/share
+    mkdir -p home/.local
+    mkdir -p $out/share
+    ln -s $out/share home/.local/share
+    HOME=$(readlink -f home) $out/bin/paperwork-shell install
```

What about the following instead?

```suggestion
    XDG_DATA_HOME="$out/share" "$out/bin/payperwork-shell" install
```

Also, did you check whether payperwork-shell actually works at runtime?

---

yes your suggestion works :)
(minus the typo)

about paperwork-shell: yes, for example `result/bin/paperwork-shell chkdeps paperwork` works whereas it used to complain about a missing paperwork module.